h/services/annotation_write.py

```diff
@@ -77,31 +77,99 @@ def create_annotation(self, data: dict) -> Annotation:
 
         return annotation
 
-    def _validate_group(self, annotation: Annotation):
+    def update_annotation(
+        # pylint: disable=too-many-arguments
+        self,
+        annotation: Annotation,
+        data: dict,
+        update_timestamp: bool = True,
+        reindex_tag: str = "storage.update_annotation",
+        enforce_write_permission: bool = True,
```

Comment:
I'd like to move the migration stuff to this service, at which point a lot of these args would become private.